K8SPG-66 PG db chart (#85)
percona/percona-helm-charts: charts/pg-db/README.md

# pg-db: A chart for installing Percona Distribution for PostgreSQL Operator managed Databases

> This chart implements Percona Distribution for PostgreSQL deployment in Kubernets via Custom Resource object. The project itself can be found here:

Reviewer: ```suggestion
This chart implements Percona Distribution for PostgreSQL deployment on Kubernetes via a Custom Resource object. The project itself can be found here:
```

Author: done

* <https://github.com/percona/percona-postgresql-operator>

## Pre-requisites

> * [PG operator](https://hub.helm.sh/charts/percona/pg-operator) running in you K8S cluster

Reviewer: ```suggestion
* [Percona Distribution for PostgreSQL Operator](https://hub.helm.sh/charts/percona/pg-operator) running in you K8S cluster
```

Author: done

To install the chart with the `pg` release name using a dedicated namespace (recommended):

```sh
helm repo add percona https://percona.github.io/percona-helm-charts/
helm install my-db percona/pg-db --version 0.2.0 --namespace my-namespace
```

The chart can be customized using the following configurable parameters:

| Parameter                       | Description                                                                   | Default                                   |
| ------------------------------- | ------------------------------------------------------------------------------| ------------------------------------------|
| `pause`                         | Stop PostgreSQL Database safely                                                    | `false`                                   |
| `standby`                         | Switch/start PostgreSQL Database in standby mode                                                    | `false`                                   |
| `disableAutofail`                         | Disables the high availability capabilities of a PostgreSQL cluster                                                    | `false`                                   |

> | `keepData`                         | Keep database container PVC intact after cluster removal                                                    | `false`                                   |

Reviewer: ```suggestion
| `keepData`                         | Keep database container PVC intact after the cluster removal                                                | `false`                                   |
```

Author: done

> | `keepBackups`                         | Keep backups intact after cluster removal                                                    | `false`                                   |

Reviewer: ```suggestion
| `keepBackups`                         | Keep backups intact after the cluster removal                                                | `false`                                   |
```

Author: done

> | `restoreFrom`                         | Set cluster/backup name as data source for newly created cluster                                                    | `false`                                   |

Reviewer: ```suggestion
| `restoreFrom`                         | Set cluster/backup name as data source for a newly created cluster                                                  | `false`                                   |
```

Author: done

| `restoreOpts`                         | Set restore (`pgBackrest`) options for restore action                                                    | `false`                                   |
| `defaultUser`                         | Default unprivileged database username                                                    | `pguser`                                   |

> | `defaultDatabase`                         | Default user database created from cluster start                                                    | `pgdb`                                   |

Reviewer: ```suggestion
| `defaultDatabase`                         | Default user database created from the cluster start                                                | `pgdb`                                   |
```

Author: done

| `backup.volumeSpec.storagetype`                     | PostgreSQL pgBackrest container PVC storagetype | `dynamic`                              |
| `backup.volumeSpec.storageclass`                     | PostgreSQL pgBackrest container PVC storageclass | `standard`                              |
| `backup.storages.storage0.bucket`                     | `Should be stated explicitly` Storage bucket |``|
| `backup.storages.storage0.name`                     | `Should be stated explicitly for S3` S3-compatible storage name |``|
| `backup.storages.storage0.type`                     | `Should be stated explicitly for S3` S3-compatible storage type |`s3`|
| `backup.storages.storage0.endpointUrl`              | `Should be stated explicitly for S3` S3-compatible storage endpoint |``|
| `backup.storages.storage0.region`                     | `Should be stated explicitly for S3` S3-compatible storage region |``|
| `backup.storages.storage0.uriStyle`                     | (Optional) S3-compatible storage URI style |`path`|
| `backup.storages.storage0.verifyTLS`                     | (Optional) S3-compatible storage URI style |`true`|

> | `pgBouncer.image`                     | Set this variable if you need to use custom pgbouncer image                                        | `percona/percona-postresql-operator:0.2.0-ppg13-pgbouncer`                              |

Reviewer: ```suggestion
| `pgBouncer.image`                     | Set this variable if you need to use a custom pgbouncer image                                      | `percona/percona-postresql-operator:0.2.0-ppg13-pgbouncer`                              |
```

Author: done

| `pgBouncer.size`                     | The number of pgbouncer instanses                                        | `1`                              |
| `pgBouncer.resources.requests.cpu`                     | Container resource request limits CPU                                        | `1`                              |
| `pgBouncer.resources.requests.memory`                     | Container resource request limits RAM                                        | `128Mi`                              |
| `pgBouncer.resources.limits.cpu`                     | Container resource limits CPU                                        | `2`                              |
| `pgBouncer.resources.limits.memory`                     | Container resource limits RAM                                        | `512Mi`                              |

> | `pgBouncer.expose.serviceType`                     | K8S service type for pgbouncer deployment | `ClusterIP`                              |

Reviewer: ```suggestion
| `pgBouncer.expose.serviceType`                     | K8S service type for the pgbouncer deployment | `ClusterIP`                              |
```

Author: done

| `pgBadger.enabled` | Switch on pgBadger | `false` |
| `pgBadger.image`              | pgBadger image                                            | `percona/percona-postgresql-operator:0.2.0-ppg13-pgbadger` |
| `pgBadger.port`                     | pgBadger port                                       | `10000` |
| `secrets.primaryuser`                     | primary user password (in use for replication only)                                       |`autogrenerated by operator`|
| `secrets.postgres`                     | postges user password (superuser, not accessible via pgbouncer)                                       |`autogrenerated by operator`|
| `secrets.pgbouncer`                     | pgbouncer user password                                        |`autogrenerated by operator`|
| `secrets.<default_user>`                     | Default user password                                        |`autogrenerated by operator`|

Specify parameters using `--set key=value[,key=value]` argument to `helm install`
Notice that you can use multiple replica sets only with sharding enabled.

## Examples

> ### Deploy a replica set with disabled backups and no mongos pods

Reviewer: mongos ?

Author: done

This is great for a dev Percona Distribution for PostgreSQL cluster as it doesn't bother with backups.

```bash
$ helm install dev  --namespace pgdb .
NAME: dev
LAST DEPLOYED:
NAMESPACE: pgdb
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
```

> You can start up cluster with only S3 backup storage like this

Reviewer: ```suggestion
You can start up the cluster with only S3 backup storage like this
```

Author: done
